gcc/rust/rust-gcc-diagnostics.cc

```diff
@@ -70,6 +71,38 @@ rust_be_error_at (const RichLocation &location, const std::string &errmsg)
   error_at (&gcc_loc, "%s", errmsg.c_str ());
 }
 
+class rust_error_code_rule : public diagnostic_metadata::rule
+{
+public:
+  rust_error_code_rule (const ErrorCode code) : m_code (code) {}
+
+  char *make_description () const final override
```

Comment:
Maybe i am missing something but is this used? Just wondering should it just use a std::string instead of relying on xstrdup which will leak.